services/server/src/server/services/storageServices/RepositoryV2Service.ts

```diff
@@ -0,0 +1,580 @@
+import dirTree from "directory-tree";
+import Path from "path";
+import fs from "fs";
+import {
+  Match,
+  Status,
+  StringMap,
+  /* ContextVariables, */
+  CheckedContract,
+} from "@ethereum-sourcify/lib-sourcify";
+import { MatchLevel, RepositoryTag } from "../../types";
+import {
+  create as createIpfsClient,
+  IPFSHTTPClient,
+  globSource,
+} from "ipfs-http-client";
+import path from "path";
+import { logger } from "../../../common/logger";
+import { getAddress, id as keccak256 } from "ethers";
+import { getMatchStatus } from "../../common";
+import { IStorageService } from "../StorageService";
+import config from "config";
+
+const REPOSITORY_VERSION = "0.1";
+/**
+ * A type for specifying the match quality of files.
+ */
+type MatchQuality = "full" | "partial";
+
+type FilesInfo<T> = { status: MatchQuality; files: Array<T> };
+
+interface FileObject {
+  name: string;
+  path: string;
+  content?: string;
+}
+type PathConfig = {
+  matchQuality: MatchQuality;
+  chainId: string;
+  address: string;
+  fileName?: string;
+  source?: boolean;
+};
+
+declare interface ContractData {
+  full: string[];
+  partial: string[];
+}
+
+export interface RepositoryV2ServiceOptions {
+  ipfsApi: string;
+  repositoryPath: string;
+  repositoryVersion: string;
+}
+
+interface FileObject {
+  name: string;
+  path: string;
+  content?: string;
+}
+
+declare interface ContractData {
+  full: string[];
+  partial: string[];
+}
+
```

Comment:
All these types seem to be copy pasted, some even exist in types. These should all come from a single source